src/main/java/taskon/storage/Storage.java

```diff
@@ -0,0 +1,154 @@
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            File file = new File(this.path);
+
+            // check if directory or file do not exist
+            if (!file.getParentFile().exists()) {
+                file.getParentFile().mkdirs();
+            }
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+
+            Scanner scanner = new Scanner(file);
+            while (scanner.hasNextLine()) {
+                String line = scanner.nextLine();
+                tasks.add(parseTask(line));
+            }
+            scanner.close();
+
+        } catch (FileNotFoundException | TaskonException e) {
+            System.out.println("File not found!");
+        } catch (IOException e) {
+            System.out.println("Error: " + e.getMessage());
+        }
+        return tasks;
+    }
+
+
+    /**
+     * Converts a line from the storage file into a Task object.
+     *
+     * @param line The line from the file representing a task.
+     * @return The Task object created from the line.
+     * @throws TaskonException If the line cannot be parsed into a valid Task.
+     */
+    private static Task parseTask(String line) throws TaskonException {
+        Task task = null;
+        String[] taskDescription = line.trim().split("\\s*\\" + SEPARATOR + "\\s*");
+        String taskType = taskDescription[0];
+
+        switch (taskType) {
+        case "T":
+            task = new Todo(taskDescription[2]);
+            break;
+
+        case "D":
+            task = new Deadline(taskDescription[2], taskDescription[3]);
+            break;
+
+        case "E":
+            task = new Event(taskDescription[2], taskDescription[3], taskDescription[4]);
+            break;
+
+        default:
+            System.out.println("Unknown task type: " + taskType);
+            break;
+        }
+
+        boolean isDone = taskDescription[1].equals("1");
+        if (task != null && isDone) {
+            try {
+                task.markAsDone();
+            } catch (TaskonException e) {
+                System.out.println(e.getMessage());
+            }
+
+        }
+
+        return task;
+    }
+
+    /**
+     * Converts a Task object into a formatted string suitable for saving to the file.
+     *
+     * @param task The Task object to convert.
+     * @return A formatted string representing the Task.
+     */
+    private static String taskToFileString(Task task) {
+        String taskStatus = task.getIsDone() ? "1" : "0";
+
+        String taskDescription = "";
+        if (task instanceof Todo) {
+            taskDescription = "T" + SEPARATOR + taskStatus + SEPARATOR + task.getDescription();
+        } else if (task instanceof Deadline) {
+            taskDescription = "D" + SEPARATOR + taskStatus + SEPARATOR + task.getDescription()
+                    + SEPARATOR + ((Deadline) task).getDate().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"));
```

Comment:
Line length is beyond the soft limit, you may want to consider adding a line break